src/core/include/openvino/op/util/weightless_caching_attributes.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/core_visibility.hpp"
+#include "openvino/core/runtime_attribute.hpp"
+
+namespace ov {
+
+/**
+ * @brief ConstantOriginalSize class represents runtime info attribute that marks
+ * the original size of the constant in bytes. It is used in the weightless caching
+ * mechanism. It's not copyable in case the data was changed (the original node was
+ * replaced by a new one produced during the tranformation pipeline) - in that case
+ * weightless caching can't be used for that constant.
+ */
+class OPENVINO_API ConstantOriginalSize : public RuntimeAttribute {
```

Comment:
Move this attribute to dev API.
It looks like for now can OV internal attribute.

Reply:
Do I move to dev API just by deleting OPENVINO_API or replacing it with something else? I didn't find any preprocessor define which has DEV.*API in the name and if I delete OPENVINO_API I get linker errors while building program_builder.cpp.

Reply:
Move header to `src/core/dev_api/openvino/core/rt_info/*.hpp`, keep `OPENVINO_API` macro.

Reply:
Done.